### PR TITLE
Fix Nova server diagnostic tests

### PR DESCRIPTION
**Pull request by bogdando**

### CODE REVIEW

**Comment by bogdando** [comment]
This fixes the following mishap
```
"+ oc exec -t openstackclient -- openstack server list",
"+ grep -qF '| test | ACTIVE |'",
"+ oc exec -t openstackclient -- openstack server stop test",
"+ oc exec -t openstackclient -- openstack server list",
"+ grep -qF '| test | SHUTOFF |'",
"+ oc exec -t openstackclient -- openstack server --os-compute-api-version 2.48 show --diagnostics test",
"+ grep 'it is in power state shutdown'",
"No ServerDiagnostics found for None: Server Error for url: https://nova-public-openstack.apps-crc.testing/v2.1/servers/0c9c00a0-95ac-4b07-a546-46f5aa59b251/diagnostics, The server didn't respond in time.: 504 Gateway Time-out",
"command terminated with exit code 1",
"+ echo PASS"], "stdout": "PASS",
"stdout_lines": ["PASS"]}
```

**Review by jistr — APPROVED** [review]

**Comment by bogdando** [comment]
this fails as the VM workload appears to be stopped initially after adoption (a regression?)
```
"stderr": "+ alias 'openstack=oc exec -t openstackclient -- openstack'
+ FIP=192.168.122.20
+ grep -qF '| test | ACTIVE |'
+ oc exec -t openstackclient -- openstack server list
+ echo FAIL
+ oc exec -t openstackclient -- openstack server stop test
Cannot 'stop' instance d0c0def7-6f63-4d1c-b46b-744f1c077e84 while it is in vm_state stopped (HTTP 409) (Request-ID: req-eda3aa87-15e4-4705-8c74-5f2ad9e649e6)
command terminated with exit code 1
+ echo FAIL
+ oc exec -t openstackclient -- openstack server list
+ grep -qF '| test | SHUTOFF |'
+ oc exec -t openstackclient -- openstack server --os-compute-api-version 2.48 show --diagnostics test
+ grep -q 'it is in power state shutdown'
+ echo FAIL",
"stderr_lines": ["+ alias 'openstack=oc exec -t openstackclient -- openstack'",
"+ FIP=192.168.122.20",
"+ grep -qF '| test | ACTIVE |'",
"+ oc exec -t openstackclient -- openstack server list",
"+ echo FAIL",
"+ oc exec -t openstackclient -- openstack server stop test",
"Cannot 'stop' instance d0c0def7-6f63-4d1c-b46b-744f1c077e84 while it is in vm_state stopped (HTTP 409) (Request-ID: req-eda3aa87-15e4-4705-8c74-5f2ad9e649e6)",
"command terminated with exit code 1",
"+ echo FAIL",
"+ oc exec -t openstackclient -- openstack server list",
"+ grep -qF '| test | SHUTOFF |'",
"+ oc exec -t openstackclient -- openstack server --os-compute-api-version 2.48 show --diagnostics test",
"+ grep -q 'it is in power state shutdown'",
"+ echo FAIL"], "stdout": "FAIL
FAIL
FAIL",
"stdout_lines": ["FAIL",
"FAIL",
"FAIL"]}
```

**Comment by bogdando** [comment]
the root cause
standalone.localdomain virtqemud[84623]: Cannot create daemon common directory '/run/libvirt/common': Not a directory

**Comment by bogdando** [comment]
recheck dep

**Comment by bogdando** [comment]
recheck dep

**Comment by bogdando** [comment]
The dependency fix https://github.com/openstack-k8s-operators/edpm-ansible/pull/599 should have helped, but I am still seeing the same problem:
```
TASK [osp.edpm.edpm_libvirt : Check if /run/libvirt has container_file_t selinux label] ***
Wednesday 13 March 2024  16:59:53 +0000 (0:00:13.573)       0:02:47.401 ******* 
ok: [standalone]

TASK [osp.edpm.edpm_libvirt : Restore selinux label for /run/libvirt to virt_var_run_t] ***
Wednesday 13 March 2024  16:59:53 +0000 (0:00:00.603)       0:02:48.004 ******* 
skipping: [standalone]
```

**Comment by bogdando** [comment]
recheck

**Comment by bogdando** [comment]
I believe the selinux relabling fix doesn't help as we got a pack of another fundamental problems related to edpm role that reboots standalone VMs during adoption (and this leaves VM shutoff, so power diag steps fail as report differently - cannot power off node in suut off state):

* we need to avoid rebooting nodes at all during adoption
* we need to instantly recover VMs, if reboot is inevitable
* we need to fix the normal VM power state recovery process, which doesn't work neither

**Comment by bogdando** [comment]
recheck dep

**Comment by bogdando** [comment]
tested, should be OK now!

**Comment by bogdando** [comment]
The depends-on change wasn't included in CI build, there was old code running:
```
TASK [dataplane_adoption : verify if Nova services can stop the existing test VM instance] ***
FAILED - RETRYING: [localhost]: verify if Nova services can stop the existing test VM instance (10 retries left).
FAILED - RETRYING: [localhost]: verify if Nova services can stop the existing test VM instance (9 retries left).
FAILED - RETRYING: [localhost]: verify if Nova services can stop the existing test VM instance (8 retries left).
FAILED - RETRYING: [localhost]: verify if Nova services can stop the existing test VM instance (7 retries left).
FAILED - RETRYING: [localhost]: verify if Nova services can stop the existing test VM instance (6 retries left).
FAILED - RETRYING: [localhost]: verify if Nova services can stop the existing test VM instance (5 retries left).
FAILED - RETRYING: [localhost]: verify if Nova services can stop the existing test VM instance (4 retries left).
FAILED - RETRYING: [localhost]: verify if Nova services can stop the existing test VM instance (3 retries left).
FAILED - RETRYING: [localhost]: verify if Nova services can stop the existing test VM instance (2 retries left).
FAILED - RETRYING: [localhost]: verify if Nova services can stop the existing test VM instance (1 retries left).
fatal: [localhost]: FAILED! => {"attempts": 10, "changed": true, "cmd": "set -euxo pipefail

alias openstack=\"oc exec -t openstackclient -- openstack\"
FIP=192.168.122.20

${BASH_ALIASES[openstack]} server list | grep -qF '| test | ACTIVE |' || echo FAIL
${BASH_ALIASES[openstack]} server stop test || echo FAIL
${BASH_ALIASES[openstack]} server list | grep -qF '| test | SHUTOFF |' || echo FAIL
${BASH_ALIASES[openstack]} server --os-compute-api-version 2.48 show --diagnostics test | grep -q \"it is in power state shutdown\" || echo FAIL
",```

**Comment by bogdando** [comment]
pj-rehearse adoption-docs-preview

File: docs_user/modules/openstack-edpm_adoption.adoc [hunk 1]
```diff
@@ -622,15 +622,14 @@ oc exec -it nova-cell1-conductor-0 -- nova-manage db online_data_migrations
...
+${BASH_ALIASES[openstack]} server --os-compute-api-version 2.48 show --diagnostics test 2>&1 || echo PASS
 ----
 
 * Verify if Nova services can start the existing test VM instance:
 +
 ----
-${BASH_ALIASES[openstack]} server list | grep -qF '| test | SHUTOFF |' && openstack server start test
-${BASH_ALIASES[openstack]} server list | grep -F '| test | ACTIVE |'
-${BASH_ALIASES[openstack]} server --os-compute-api-version 2.48 show --diagnostics test --fit-width -f json | jq -r '.state' | grep running
-----
```
Comment (jistr) [inline docs_user/modules/openstack-edpm_adoption.adoc:636]: Ah the breakage probably wasn't in the Ceph docs, it's here

Comment (bogdando) [reply docs_user/modules/openstack-edpm_adoption.adoc:636]: what is the breakage here?

Comment (bogdando) [reply docs_user/modules/openstack-edpm_adoption.adoc:636]: nvm, got it :) got to add ``----`` back, right?

**Review by cescgina — APPROVED** [review]

**Pull request merged**